src/inmanta/protocol/common.py

```diff
@@ -776,6 +794,22 @@ def build_call(self, args: List, kwargs: Dict[str, Any] = {}) -> Request:
 
         return Request(url=url, method=self.operation, headers=headers, body=body)
 
+    def _encode_dict_for_get(
+        self, query_param_name: str, query_param_value: Dict[str, Union[Any, List[Any]]]
+    ) -> Dict[str, str]:
+        """ Dicts are encoded in the following manner: param = {'ab': 1, 'cd': 2} to param.abc=1&param.cd=2 """
+        sub_dict = {}
+        for key, value in query_param_value.items():
+            if isinstance(value, list):
+                sub_dict[f"{query_param_name}.{key}"] = self._encode_list_for_get(value)
+            else:
+                sub_dict[f"{query_param_name}.{key}"] = value
```

Comment:
this is not URL encoded? 

(what about names with '.' in them? Do we cleanly reject or accept them?)

Reply:
`.` is accepted, regardless of whether it's part of the key or value. If the key inside a dict is just a dot (`param.=value`), it's rejected, but can contain them otherwise. (There is a test case for it)